File: RE/58-Length-of-Last-Word.py
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        #using split
        s = s.strip()
        if not s:
            return 0
        lst = s.split(' ')
        print lst
        return len(lst[-1])
        
        
        '''
        #Using Python String Funcs
        s = s.strip()
        lastIndex = s.rfind(' ')+1
        print lastIndex-1
        return len(s)-lastIndex

File: RE/58-Length-of-Last-Word.java
//https://leetcode.com/problems/length-of-last-word/
/*
Given a string s consists of upper/lower-case alphabets and empty space characters ' ', return the length of last word in the string.

If the last word does not exist, return 0.

Note: A word is defined as a character sequence consists of non-space characters only.

For example, 
Given s = "Hello World",
return 5.
*/
public class Solution {
    //using java regex
    public int lengthOfLastWord(String s) {
        s = s.trim();
        int lastIndex = s.lastIndexOf(' ')+1;
        //System.out.format("lastIndex is %d \n", lastIndex-1);
        //"a" lastIndex of ' ' will be -1
        return s.length()-lastIndex;
    }
}
